packages/compass-shell/src/components/compass-shell/tab-compass-shell.tsx

```diff
@@ -105,6 +108,33 @@ export const CompassShell: React.FC<CompassShellProps> = ({
     initialInput ?? ''
   );
 
+  const [mongoshVersion, setMongoshVersion] = useTabState(
+    'mongoshVersion',
+    _mongoshVersion
+  );
+  const logger = useLogger('COMPASS-SHELL');
+
+  useEffect(() => {
+    const { log, mongoLogId } = logger;
+    if (!runtime || mongoshVersion) return;
+    runtime
+      .evaluate('version()')
+      .then((value) => {
+        setMongoshVersion((_mongoshVersion = 'v' + String(value.printable)));
+      })
+      .catch((err) => {
+        setMongoshVersion('[unknown]');
+        log.error(
+          mongoLogId(1_001_000_342),
+          'shell',
+          'could not evaluate mongosh version',
+          {
+            error: (err as Error).message,
+          }
+        );
+      });
+  }, [runtime, logger, mongoshVersion, setMongoshVersion]);
+
```

Comment:
Not ideal, if you ask me, and happy to take other suggestions – I'd like to avoid adding an extra dependency on `shell-api` though, considering that in `node-runtime-worker-thread` it's a pre-bundled package

Reply:
I think this adjustment needs to happen eventually, but really don't feel strongly that it needs to happen now, we'd probably want to sketch some state management changes here for this first anyway and it's really starts being out of scope. @lerouxb is right that effects are a foot gun so the less, the better, but for now it's really pretty good and I don't see any missed logic there.

Mostly just wanted to double-check that I'm understanding correctly why we're doing the switch, so all good from my side!